### PR TITLE
dev: apply cairo-rs-py-monkeypatch to use rust VM

### PR DESCRIPTION
**Pull request by ftupas**
Time spent on this PR: 0.1

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Currently we are using the Python VM when we run our integration tests

## What is the new behavior?

- add rust bindings `cairo-rs-py`
- Use a monkey patch to override entrypoints and use the rust VM instead of Python VM

### CODE REVIEW

File: tests/fixtures/starknet.py [hunk 1]
```diff
@@ -33,6 +34,10 @@
 
 @pytest_asyncio.fixture(scope="session")
 async def starknet(worker_id, request, blockhashes) -> AsyncGenerator[Starknet, None]:
+    # apply cairo-rs-py monkey patch to use the rust VM
+    cairo_rs_py_monkeypatch()
```
Comment (ftupas) [inline tests/fixtures/starknet.py:38]: Currently failling

```13 failed, 24 passed, 2 skipped, 1 xfailed in 177.30s (0:02:57)```

Comment (ClementWalter) [reply tests/fixtures/starknet.py:38]: tests are failing on the `unsigned_div_rem` bugs of the cairo vm solved few months ago; the embedded version in cairo-rs-py needs to be updated

Comment (ftupas) [reply tests/fixtures/starknet.py:38]: Gotcha 🤙 should I make an issue to fork/pr on cairo-rs-py to update this?

**Comment by ClementWalter** [comment]
@Eikix why did you close this? it's still relevant!

**Comment by Eikix** [comment]
Interesting!

> @Eikix why did you close this? it's still relevant!

Did not know

**Comment by ClementWalter** [comment]
> Interesting!
> 
> > @Eikix why did you close this? it's still relevant!
> 
> Did not know

depending on the outcome of the discussion with LC cc @greged93 but the topic of running the unit test against the rust vm is still on

**Comment by ClementWalter** [comment]
cc @greged93 closing because we won't push this after a call with LC

**Pull request closed without merging**